### PR TITLE
Update account tests to sepolia

### LINKED ISSUE #544: goerli depreceation
Any tests and examples that currently use Goerli should use another test network, like Sepolia. 

1. Enter the rpc directory
2. rename .env.template to .env.testnet, and enter the Sepolia endpoint
3. go test ./... -env testnet

You will see a bunch of tests that currently depend on Goerli, but should be updated to depend on Sepolia instead.

Free RPC endpoints https://data.voyager.online/

### PR DESCRIPTION
**Pull request by AryanGodara**
Partially resolves #544

### CODE REVIEW

File: account/account_test.go [hunk 1]
```diff
@@ -89,37 +89,39 @@ func TestTransactionHashInvoke(t *testing.T) {
...
-					Nonce:   utils.TestHexToFelt(t, "0x2"),
-					MaxFee:  utils.TestHexToFelt(t, "0x574fbde6000"),
+					Nonce:   utils.TestHexToFelt(t, "0x3cf"),
+					MaxFee:  utils.TestHexToFelt(t, "0x1a6f9d0dc5952"),
 					Version: rpc.TransactionV1,
 				},
 			},
 			{
-				ExpectedHash:   utils.TestHexToFelt(t, "0x135c34f53f8b7f59efd450eb689fccd9dd4cfe7f9d9dc4d09954c5653138698"),
+				ExpectedHash:   utils.TestHexToFelt(t, "0x585419091c2f819470e30437d284b7507cb794c695fa92f801cd0a14f75483a"),
```
Comment (rianhughes) [inline account/account_test.go]: Would you mind removing this test?

File: account/account_test.go [hunk 2]
```diff
@@ -412,15 +414,15 @@ func TestSignMOCK(t *testing.T) {
...
+				Address:    utils.TestHexToFelt(t, "0x06fb2806bc2564827796e0796144f8104581acdcbcd7721615ad376f70baf87d"),
 				PrivKey:    utils.TestHexToFelt(t, "0x043b7fe9d91942c98cd5fd37579bd99ec74f879c4c79d886633eecae9dad35fa"),
-				ChainId:    "SN_GOERLI",
-				FeltToSign: utils.TestHexToFelt(t, "0x73cf79c4bfa0c7a41f473c07e1be5ac25faa7c2fdf9edcbd12c1438f40f13d8"),
+				ChainId:    "SN_SEPOLIA",
+				FeltToSign: utils.TestHexToFelt(t, "0x5d307ad21a407ab6e93754b2fca71dd2d3b28313f6e844a7f3ecc404263a406"),
 				ExpectedSig: []*felt.Felt{
-					utils.TestHexToFelt(t, "0x10d405427040655f118bc8b897e2f2f8147858bbcb0e3d6bc6dfbc6d0205e8"),
-					utils.TestHexToFelt(t, "0x5cdfe4a3d5b63002e9011ec0ba59ae2b75a43cb2a3bc1699b35aa64cb9ca3cf"),
+					utils.TestHexToFelt(t, "0x18efb47edd79bc20e35bdc07f71c49d1d6dc5983034e31dd111f164dfaa315d"),
```
Comment (rianhughes) [inline account/account_test.go]: Where did you get this? It doesn't match the signatures from the block explorer

Comment (rianhughes) [reply account/account_test.go]: Note, you may need to generate a new transaction with the given test private key (you should be able to use the one provided in the code - or just create a test wallet)

File: account/account_test.go [hunk 3]
```diff
@@ -478,10 +480,10 @@ func TestAddInvoke(t *testing.T) {
 		"devnet": {},
 		"testnet": {
 			{
-				// https://goerli.voyager.online/tx/0x73cf79c4bfa0c7a41f473c07e1be5ac25faa7c2fdf9edcbd12c1438f40f13d8#overview
+				// https://sepolia.voyager.online/tx/0x5d307ad21a407ab6e93754b2fca71dd2d3b28313f6e844a7f3ecc404263a406#overview
```
Comment (rianhughes) [inline account/account_test.go]: The testdata from these cases needs to come from the Sepolia network. I would suggest creating it using a test wallet

File: account/account_test.go [hunk 4]
```diff
@@ -766,10 +769,11 @@ func TestTransactionHashDeclare(t *testing.T) {
 					NonceDataMode:         rpc.DAModeL1,
 					FeeMode:               rpc.DAModeL1,
 				},
-				ExpectedHash: utils.TestHexToFelt(t, "0x41d1f5206ef58a443e7d3d1ca073171ec25fa75313394318fc83a074a6631c3"),
+				ExpectedHash: utils.TestHexToFelt(t, "0x61bfaf480ac824971ad1bdc316fa821f58afd6b47e037242ef265d0aaea7c78"),
```
Comment (rianhughes) [inline account/account_test.go]: The test data from TestTransactionHashDeclare also needs to come from the Sepolia network

File: account/account_test.go [hunk 5]
```diff
@@ -883,11 +887,11 @@ func TestTransactionHashdeployAccount(t *testing.T) {
 					},
 				},
 				SenderAddress: utils.TestHexToFelt(t, "0x88d0038623a89bf853c70ea68b1062ccf32b094d1d7e5f924cda8404dc73e1"),
-				ExpectedHash:  utils.TestHexToFelt(t, "0x5b6b5927cd70ad7a80efdbe898244525871875c76540b239f6730118598b9cb"),
+				ExpectedHash:  utils.TestHexToFelt(t, "0x41ffcaa2f472e58b496f58aabae0590d4b2cf6117201c988d51c3f040e489e6"),
```
Comment (rianhughes) [inline account/account_test.go]: This also doesn't seem to exist on the sepolia network.

File: account/account_test.go [hunk 6]
```diff
@@ -1079,7 +1083,7 @@ func TestWaitForTransactionReceipt(t *testing.T) {
 //
 //	none
 func TestAddDeclareTxn(t *testing.T) {
-	// https://goerli.voyager.online/tx/0x76af2faec46130ffad1ab2f615ad16b30afcf49cfbd09f655a26e545b03a21d
+	// https://sepolia.voyager.online/tx/0x76af2faec46130ffad1ab2f615ad16b30afcf49cfbd09f655a26e545b03a21d
```
Comment (rianhughes) [inline account/account_test.go]: This transaction doesn't exist on the Sepolia network.

**Comment by rianhughes** [comment]
Just two general points:
1. You need to replace the test data that was gathered from the Goerli network, with new test data from the Sepolia network. This will go for the other PRs as well.

2. I would suggest running the tests in the account package against a Sepolia network endpoint, ie running "go test ./... -env testnet -v" from the accounts pacakge. You will probably need to put a .env.tesnet file in the account package, with a URL of a Seploia full node. https://data.voyager.online/

**Comment by AryanGodara** [comment]
> Just two general points:
> 
> 1. You need to replace the test data that was gathered from the Goerli network, with new test data from the Sepolia network. This will go for the other PRs as well.
> 2. I would suggest running the tests in the account package against a Sepolia network endpoint, ie running "go test ./... -env testnet -v" from the accounts pacakge. You will probably need to put a .env.tesnet file in the account package, with a URL of a Seploia full node. https://data.voyager.online/

I used existing transactions which were similar to the goerli ones. So these were real transactions I found on the block-explorer. But some of them don't seem to exist anymore. (I'd picked the most recent ones that matched) For the other tests also, I've used real sepolia data.

But you're right. I need to create new transactions here, with a test wallet. I'll get to this asap!

File: account/account_test.go [hunk 7]
```diff
@@ -723,21 +655,22 @@ func TestTransactionHashDeclare(t *testing.T) {
...
-				MaxFee:            utils.TestHexToFelt(t, "0x50c8f3053db"),
-				Type:              rpc.TransactionType_Declare,
-				Version:           rpc.TransactionV2,
-				Signature:         []*felt.Felt{},
-				SenderAddress:     utils.TestHexToFelt(t, "0x36437dffa1b0bf630f04690a3b302adbabb942deb488ea430660c895ff25acf"),
-				CompiledClassHash: utils.TestHexToFelt(t, "0x615a5260d3d47d79fba87898da95cb5394b181c7d5097bc8ced4ed06ac24ac5"),
-				ClassHash:         utils.TestHexToFelt(t, "0x639cdc0c42c8c4d3d805e56294fa0e6bf5a584ad0fcd538b843cc294913b982"),
+		"mock": {
+			{
+				// Note this is a testnet / sepolia transaction
```
Comment (rianhughes) [inline account/account_test.go]: This is a goerli txn

File: account/account_test.go [hunk 8]
```diff
@@ -723,21 +655,22 @@ func TestTransactionHashDeclare(t *testing.T) {
...
+					ClassHash:         utils.TestHexToFelt(t, "0x639cdc0c42c8c4d3d805e56294fa0e6bf5a584ad0fcd538b843cc294913b982"),
+				},
+				ExpectedHash: utils.TestHexToFelt(t, "0x31e1897656b75c64997a6c2e1a6b23d46866656ceff671f4a947a09b08fcb43"),
+				ExpectedErr:  nil,
 			},
-			ExpectedHash: utils.TestHexToFelt(t, "0x4e0519272438a3ae0d0fca776136e2bb6fcd5d3b2af47e53575c5874ccfce92"),
-			ExpectedErr:  nil,
-		},
 			{
 				// https://external.integration.starknet.io/feeder_gateway/get_transaction?transactionHash=0x41d1f5206ef58a443e7d3d1ca073171ec25fa75313394318fc83a074a6631c3
```
Comment (rianhughes) [inline account/account_test.go]: this is also goerli I think

File: account/account_test.go [hunk 9]
```diff
@@ -795,7 +729,7 @@ func TestTransactionHashInvokeV3(t *testing.T) {
 	testSet := map[string][]testSetType{
 		"mock": {
 			{
-				// https://external.integration.starknet.io/feeder_gateway/get_transaction?transactionHash=0x49728601e0bb2f48ce506b0cbd9c0e2a9e50d95858aa41463f46386dca489fd
+				// https://external.integration.starknet.io/feeder_gateway/get_transaction?transactionHash=0x473e7d9dcaf02b5bbc2f89486fb2e23241f645cd55f70ff02576b79541befe7
```
Comment (rianhughes) [inline account/account_test.go]: and this

File: account/account_test.go [hunk 10]
```diff
@@ -883,11 +817,11 @@ func TestTransactionHashdeployAccount(t *testing.T) {
 					},
 				},
 				SenderAddress: utils.TestHexToFelt(t, "0x88d0038623a89bf853c70ea68b1062ccf32b094d1d7e5f924cda8404dc73e1"),
-				ExpectedHash:  utils.TestHexToFelt(t, "0x5b6b5927cd70ad7a80efdbe898244525871875c76540b239f6730118598b9cb"),
+				ExpectedHash:  utils.TestHexToFelt(t, "0x41ffcaa2f472e58b496f58aabae0590d4b2cf6117201c988d51c3f040e489e6"),
 				ExpectedErr:   nil,
 			},
 			{
-				// https://external.integration.starknet.io/feeder_gateway/get_transaction?transactionHash=0x29fd7881f14380842414cdfdd8d6c0b1f2174f8916edcfeb1ede1eb26ac3ef0
+				// https://external.integration.starknet.io/feeder_gateway/get_transaction?transactionHash=0x3018236df5779c1f28caba0e64febcb78f5bc69aa3538be54f4e27def9de1b3
```
Comment (rianhughes) [inline account/account_test.go]: the test data in this also seems to be from goerli

File: account/account_test.go [hunk 11]
```diff
@@ -90,45 +90,31 @@ func TestTransactionHashInvoke(t *testing.T) {
...
+						"0xcaffbd1bd76bd7f24a3fa1d69d1b2588a86d1f9d2359b13f6a84b7e1cbd126",
+						"0x7",
+						"0x457874726163745265736f7572636546696e697368",
+						"0x5",
+						"0x5",
+						"0xb82",
 						"0x1",
-						"0x0",
+						"0x1",
+						"0x35c",
```
Comment (rianhughes) [inline account/account_test.go:112]: This calldata doesn't seem to match what's on Voayger

File: account/account_test.go [hunk 12]
```diff
@@ -440,8 +438,8 @@ func TestSignMOCK(t *testing.T) {
 		account, err := account.NewAccount(mockRpcProvider, test.Address, test.Address.String(), ks, 0)
 		require.NoError(t, err, "error returned from account.NewAccount()")
 
-		msg := utils.TestHexToFelt(t, "0x73cf79c4bfa0c7a41f473c07e1be5ac25faa7c2fdf9edcbd12c1438f40f13d8")
-		sig, err := account.Sign(context.Background(), msg)
+		// msg := utils.TestHexToFelt(t, "0x5d307ad21a407ab6e93754b2fca71dd2d3b28313f6e844a7f3ecc404263a406")
```
Comment (rianhughes) [inline account/account_test.go]: We can delete this line

File: account/account_test.go [hunk 13]
```diff
@@ -478,123 +476,57 @@ func TestAddInvoke(t *testing.T) {
...
-					Calldata:           []*felt.Felt{utils.TestHexToFelt(t, "0xaC25b2B9F4ca06179fA0D2522F47Bc86A9DF9314")},
-				},
-			},
 		},
 		"mainnet": {},
 	}[testEnv]
 
-	for _, test := range testSet {
+	for i, test := range testSet {
+		fmt.Println("Test case", i)
```
Comment (rianhughes) [inline account/account_test.go]: We can also delete this

File: account/account_test.go (hunk 11 above)
Comment (AryanGodara) [reply account/account_test.go:112]: image
The `Calls` section of the calldata doesn't include the entire calldata for the transaction. Switching to the `transaction` tab shows the entire thing.
(Just noticed this, I'd been using direct RPC calls, so just checked this was the case)

File: account/account_test.go [hunk 14]
```diff
@@ -141,8 +141,60 @@ func TestTransactionHashInvoke(t *testing.T) {
...
+						"0x52884ee3f",
+					}),
+				},
+				TxDetails: rpc.TxDetails{
+					Nonce:   utils.TestHexToFelt(t, "0x1"),
+					MaxFee:  utils.TestHexToFelt(t, "0x2a173cd36e400"),
+					Version: rpc.TransactionV1,
+				},
+			},
+		},
```
Comment (rianhughes) [inline account/account_test.go]: This isn't a sepolia tx, right? If not, we can just remove it

File: rpc/types_broadcast_transaction.go [hunk 15]
```diff
@@ -53,28 +53,28 @@ type BroadcastDeclareTxnV1 struct {
 	// SenderAddress the address of the account contract sending the declaration transaction
 	SenderAddress *felt.Felt              `json:"sender_address"`
 	MaxFee        *felt.Felt              `json:"max_fee"`
-	Version       NumAsHex                `json:"version"`
```
Comment (thiagodeev) [inline rpc/types_broadcast_transaction.go:56]: These changes were updated months ago
https://github.com/starkware-libs/starknet-specs/blob/master/api/starknet_api_openrpc.json#L1884

File: account/account_test.go [hunk 16]
```diff
@@ -723,53 +650,76 @@ func TestTransactionHashDeclare(t *testing.T) {
...
+					// 	L2Gas: rpc.ResourceBounds{
+					// 		MaxAmount:       "0x0",
+					// 		MaxPricePerUnit: "0x0",
+					// 	},
+					// },
+					// Tip:                   "0x0",
+					// PayMasterData:         []*felt.Felt{},
+					// AccountDeploymentData: []*felt.Felt{},
+					// NonceDataMode:         rpc.DAModeL1,
+					// FeeMode:               rpc.DAModeL1,
```
Comment (rianhughes) [inline account/account_test.go]: If this isnt being used it should be deleted

File: account/account_test.go [hunk 17]
```diff
@@ -210,10 +204,11 @@ func TestFmtCallData(t *testing.T) {
 	}
 	testSet := map[string][]testSetType{
 		"devnet": {},
-		"mock": {
+		"mock":   {},
+		"testnet": {
 			{
 				CairoVersion: 0,
-				ChainID:      "SN_GOERLI",
+				ChainID:      "SN_SEPOLIA",
```
Comment (rianhughes) [inline account/account_test.go]: this testdata is from goerli, we can just remove this

File: account/account_test.go [hunk 18]
```diff
@@ -1111,23 +1185,26 @@ func TestAddDeclareTxn(t *testing.T) {
...
 	require.NoError(t, err)
 
 	var casmClass contracts.CasmClass
 	err = json.Unmarshal(content2, &casmClass)
 	require.NoError(t, err)
 	compClassHash := hash.CompiledClassHash(casmClass)
 
-	nonce, err := acnt.Nonce(context.Background(), rpc.BlockID{Tag: "latest"}, acnt.AccountAddress)
-	require.NoError(t, err)
+	// nonce, err := acnt.Nonce(context.Background(), rpc.BlockID{Tag: "latest"}, acnt.AccountAddress)
```
Comment (rianhughes) [inline account/account_test.go]: we can remove this if it's not being used

File: account/account_test.go (hunk 16 above)
Comment (thiagodeev) [reply account/account_test.go]: Done

File: account/account_test.go (hunk 1 above)
Comment (thiagodeev) [reply account/account_test.go]: Didn't find this line in the code. Hence, considered fixed.

File: account/account_test.go (hunk 2 above)
Comment (thiagodeev) [reply account/account_test.go]: Didn't find this line in the code. Hence, considered fixed.

File: account/account_test.go (hunk 3 above)
Comment (thiagodeev) [reply account/account_test.go]: Looks like is everything good with these transactions.

File: account/account_test.go (hunk 4 above)
Comment (thiagodeev) [reply account/account_test.go]: Didn't find this line in the code. Hence, considered fixed.

File: account/account_test.go (hunk 5 above)
Comment (thiagodeev) [reply account/account_test.go]: Didn't find this line in the code. Hence, considered fixed.

File: account/account_test.go (hunk 6 above)
Comment (thiagodeev) [reply account/account_test.go]: Didn't find this line in the code. Hence, considered fixed.

File: account/account_test.go (hunk 7 above)
Comment (thiagodeev) [reply account/account_test.go]: Didn't find this line in the code. Hence, considered fixed.

File: account/account_test.go (hunk 8 above)
Comment (thiagodeev) [reply account/account_test.go]: Didn't find this line in the code. Hence, considered fixed.

File: account/account_test.go (hunk 9 above)
Comment (thiagodeev) [reply account/account_test.go]: Didn't find this line in the code. Hence, considered fixed.

File: account/account_test.go (hunk 10 above)
Comment (thiagodeev) [reply account/account_test.go]: Didn't find this line in the code. Hence, considered fixed.

File: account/account_test.go (hunk 12 above)
Comment (thiagodeev) [reply account/account_test.go]: Didn't find this line in the code. Hence, considered fixed.

File: account/account_test.go (hunk 13 above)
Comment (thiagodeev) [reply account/account_test.go]: Didn't find this line in the code. Hence, considered fixed.

File: account/account_test.go (hunk 14 above)
Comment (thiagodeev) [reply account/account_test.go]: Didn't find this line in the code. Hence, considered fixed.

**Review by rianhughes — APPROVED** [review]

**Pull request merged**